src/linear/coordinate_common.h

```diff
@@ -98,16 +93,15 @@ inline std::pair<double, double> GetGradient(
  *
  * \return  The gradient and diagonal Hessian entry for a given feature.
  */
-
-inline std::pair<double, double> GetGradientParallel(
-    int group_idx, int num_group, int fidx,
-
-    const std::vector<bst_gpair> &gpair, DMatrix *p_fmat) {
+inline std::pair<double, double> GetGradientParallel(int group_idx, int num_group, int fidx,
+                                                     const std::vector<bst_gpair> &gpair,
+                                                     DMatrix *p_fmat) {
   double sum_grad = 0.0, sum_hess = 0.0;
-  dmlc::DataIter<ColBatch> *iter = p_fmat->ColIterator();
+  const std::vector<bst_uint> fset(1u, static_cast<bst_uint>(fidx));
+  dmlc::DataIter<ColBatch> *iter = p_fmat->ColIterator(fset);
```

Comment:
This can use a list initialiser to save one line:
dmlc::DataIter<ColBatch> *iter = p_fmat->ColIterator({static_cast<bst_uint>(fidx)});